OSSM-6260: Document migration from SMCP managed gateways   Part of Service Mesh 2.6 release (#78340)
openshift/openshift-docs: modules/ossm-migrate-away-from-gateways-managed-by-the-smcp-resource.adoc

In {SMProductName} 2.x, the {SMProductShortName} Operator creates an ingress and egress gateway in the control plane namespace by default. You may also define additional gateways in the `ServiceMeshControlPlane` resource. However, since the introduction of gateway injection in {SMProductShortName} 2.3, the preferred method for deploying ingress and egress gateways is to do so with a Deployment resource using gateway injection. This deployment method not only provides greater flexibility and control, but is a better practice because it allows you to manage gateways with their corresponding applications rather than in the control plane resource. Therefore, you should consider disabling the default gateways, moving away from the SMCP declaration, and begin using gateway injection.

To remove the default gateways from the control plane, disable them in the SMCP resource as follow:

.Example specification to disable gateways in the SMCP resouce
[source,yaml, subs="attributes,verbatim"]
----
apiVersion: maistra.io/v2
kind: ServiceMeshControlPlane
metadata:
namespace: istio-system
spec:
gateways:
  egress:
    enabled: false
  ingress:
    enabled: false
----

> To migrate away from an SMCP defined gateway you must first deploy your custom gateway with gateway injection. Then, create either a `Route` for the gateway or a `LoadBalancer` service, depending on how the gateway is being exposed to the mesh. Switch over the gateway usage and then disable the SMCP managed gateway. Going forward, gateway deployments can be defined as part of the application deployment.

Reviewer: ```suggestion
To migrate away from an SMCP defined gateway you must first deploy your custom gateway with gateway injection. Then, create either a `Route` for the gateway or a `LoadBalancer` service, depending on how you want the gateway to be exposed to external traffic. Switch over the gateway usage and then disable the SMCP managed gateway. Going forward, gateway deployments can be defined as part of the application deployment.
```
We also probably want to provide a little more detail than `switch over the gateway usage`... Users need to update the `spec.selector` of Istio `Gateway` resources to match the new gateway pod labels; For egress gateways, users also need to update `DestinationRules` and `VirtualServices` to use the hostname of the new egress gateway service.
